File: Practica1/Ejercicio4/filterlist.py
class FilterList:
    def __init__(self):
        self.filterList = []
        self.target = None
        
    def add_filter(self, filter):
        self.filterList.append(filter)

    def set_target(self, target):
        self.target = target

    def execute(self, credentials):
        for filter in self.filterList:
            filter.execute(credentials)
        if self.target is not None:
            self.target.print(credentials)

File: Practica2/ejercicio_conjunto_filtros/lib/FilterList.dart
import 'dart:js_interop';

import 'package:ejercicio_conjunto_filtros/Credentials.dart';
import 'package:ejercicio_conjunto_filtros/FormController.dart';
import 'package:ejercicio_conjunto_filtros/filters.dart';

class FilterList {
  List<Filter> filterList = [];
  CredentialsManager? _credentialsManager;

  void addFilter(Filter filter) {
    filterList.add(filter);
  }

  void execute(FormController credentials) {
    for (var filter in filterList) {
      filter.execute(credentials);
    }
    if (credentials.valid) {
      if (_credentialsManager==null)
        throw NullRejectionException(false);
      _credentialsManager?.register(credentials.email, credentials.password);
    }
  }

  void setTarget(CredentialsManager credentialsManager) {
    this._credentialsManager = credentialsManager;
  }
}